pkg/services/apiserver/service.go

```diff
@@ -150,6 +151,8 @@ type service struct {
 	unified         resource.ResourceClient
 
 	buildHandlerChainFuncFromBuilders builder.BuildHandlerChainFuncFromBuilders
+
+	mu sync.Mutex
```

Comment:
@toddtreece

Reply:
why is this needed?

Reply:
Any updates on this?

Reply:
I pulled the latest changes from `main` and I could remove this sync and I don't see the issue anymore 🤷🏼‍♀️